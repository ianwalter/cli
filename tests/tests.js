const { test } = require('@ianwalter/bff')
const execa = require('execa')
const exampleCli = require('./fixtures/exampleCli')

test('cli', async ({ expect }) => {
  const { stdout } = await execa('./tests/fixtures/exampleCli.js', ['-c', '4'])
  expect(stdout).toMatchSnapshot()
})

test('dot format', async ({ expect }) => {
  const args = ['--some.other.thing', 'b']
  const { stdout } = await execa('./tests/fixtures/exampleCli.js', args)
  expect(stdout).toMatchSnapshot()
})

test('default', async ({ expect }) => {
  expect(exampleCli.path).toBe('/some/path')
})
