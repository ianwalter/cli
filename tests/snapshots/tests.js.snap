// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli 1`] = `
"    
     1 {
     2   packageJson: {
     3     name: '@ianwalter/cli'
     4   },
     5   level: 'debug',
     6   processorConcurrency: 4,
     7   some: {
     8     thing: true,
     9     other: {
    10       thing: 'a'
    11     }
    12   },
    13   _: [],
    14   c: 4,
    15   path: '/some/path',
    16   debug: false,
    17   d: false
    18 }"
`;

exports[`dot format 1`] = `
"    
     1 {
     2   packageJson: {
     3     name: '@ianwalter/cli'
     4   },
     5   level: 'debug',
     6   processorConcurrency: 1,
     7   some: {
     8     thing: true,
     9     other: {
    10       thing: 'b'
    11     }
    12   },
    13   _: [],
    14   c: 1,
    15   path: '/some/path',
    16   debug: false,
    17   d: false
    18 }"
`;
